coderd/autobuild/executor/lifecycle_executor.go

```diff
@@ -209,36 +209,54 @@ func build(ctx context.Context, store database.Store, workspace database.Workspa
 	}
 	provisionerJobID := uuid.New()
 	now := database.Now()
-	newProvisionerJob, err := store.InsertProvisionerJob(ctx, database.InsertProvisionerJobParams{
-		ID:             provisionerJobID,
-		CreatedAt:      now,
-		UpdatedAt:      now,
-		InitiatorID:    workspace.OwnerID,
-		OrganizationID: template.OrganizationID,
-		Provisioner:    template.Provisioner,
-		Type:           database.ProvisionerJobTypeWorkspaceBuild,
-		StorageMethod:  priorJob.StorageMethod,
-		StorageSource:  priorJob.StorageSource,
-		Input:          input,
-	})
-	if err != nil {
-		return xerrors.Errorf("insert provisioner job: %w", err)
+
+	var buildReason database.BuildReason
+	switch trans {
+	case database.WorkspaceTransitionStart:
+		buildReason = database.BuildReasonAutostart
+	case database.WorkspaceTransitionStop:
+		buildReason = database.BuildReasonAutostop
+	default:
+		return xerrors.Errorf("Unsupported transition: %q", trans)
 	}
-	_, err = store.InsertWorkspaceBuild(ctx, database.InsertWorkspaceBuildParams{
-		ID:                workspaceBuildID,
-		CreatedAt:         now,
-		UpdatedAt:         now,
-		WorkspaceID:       workspace.ID,
-		TemplateVersionID: priorHistory.TemplateVersionID,
-		BuildNumber:       priorBuildNumber + 1,
-		Name:              namesgenerator.GetRandomName(1),
-		ProvisionerState:  priorHistory.ProvisionerState,
-		InitiatorID:       workspace.OwnerID,
-		Transition:        trans,
-		JobID:             newProvisionerJob.ID,
+
+	err = store.InTx(func(db database.Store) error {
```

Comment:
This is already in one big transaction, see line 80 -- since we're already in a tx, this ends up being a no-op

Reply:
Oh TIL. I didn't see that @AbhineetJain

Reply:
Thank @johnstcn. This was also the reason the tests were failing.

Reply:
Issue: https://github.com/coder/coder/issues/2468